pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -291,7 +293,8 @@ func computeControllerRegistrationMaps(
 	controllerRegistrationList []*gardencorev1beta1.ControllerRegistration,
 ) map[string]controllerRegistration {
 	var out = make(map[string]controllerRegistration)
-	for _, cr := range controllerRegistrationList {
+
+	for _, cr := range getActiveControllerRegistrations(controllerRegistrationList) {
```

Comment:
Hmm, if you filter out the "to-be-deleted" registrations here, the controller will always exit with error in line 165 if there is a registration with deletion timestamp but shoots/installations needing that registration.

For example, I tested this by
1. adding a networking registration of type `foo`
2. adding a shoot using this network type
3. deleting the registration

gcm will log:
```
time="2020-07-23T08:17:10+02:00" level=info msg="[CONTROLLERINSTALLATION SEED] Reconciling gcp" controllerregistration-seed=staging-gcp
time="2020-07-23T08:17:10+02:00" level=info msg="Error syncing ControllerRegistration-Seed gcp: need to install an extension controller for \"Network/foo\" but no appropriate ControllerRegistration found"
```

Also, if the shoot is deleted again and the registration is not needed anymore but already marked for deletion, the controller will not remove the installations anymore, because it will fail in line 371 (`ControllerRegistration "foo" does not exist`).

Can you fix those cases?

Reply:
Nice finding, thanks!

I was testing only the case with `os-suse-*` where there was multiple extensions handling the same resources.
I have changed it now to silently not create, but it is still updating existing controller installations.